### ISSUE TITLE
bug: IonMenu not closing when clicking IonBackdrop iOS

### ISSUE THREAD

**Issue by harryFBloch** [ionic-team/ionic#19271]
# Bug Report

**Ionic version:**
[x] **ionic/react 4.8.0-rc.1**

**Current behavior:**
Ion menu with slot="end" does not close as expected on iOS but does close when using web or android.

**Expected behavior:**
Ion menu should close when clicking the backdrop. 

**Steps to reproduce:**

Right Menu Bug
From iOS browser or running on an iOS device/ simulator.
Click the right menu button in the toolbar and then try and click on backdrop to close the menu.
works in browser but not on iOS device/simulator or iOS browser. 

**Related code:**
https://stackblitz.com/edit/ionic-v4-react-eyan3m?file=Hello.jsx

**Other information:**

works with swipe gestures but not with clicking the backdrop to close on iOS devices or iOS browser. Tested with capacitor/iOS 1.2.0 and 1.1.1

**Ionic info:**

```
Ionic:

   Ionic CLI       : 5.1.1 (/usr/local/lib/node_modules/ionic)
   Ionic Framework : @ionic/react 4.8.0-rc.1

Capacitor:

   Capacitor CLI   : 1.1.1
   @capacitor/core : 1.1.1

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v12.3.1 (/usr/local/bin/node)
   npm    : 6.9.0
   OS     : macOS Mojave

```

**Comment by liamdebeasi** [comment]
Thanks for the issue. Looks like the bug is coming from here: https://github.com/ionic-team/ionic/blob/master/core/src/components/menu/menu.tsx#L218

`shouldClose` is `false` when tapping the backdrop on a menu with `slot="end"`. We will look into this!

**Comment by calvinckho** [comment]
+1

**Comment by harryFBloch** [comment]
> Any work around? And thank you guys for your time.
Using swipe to close or I added a small x button in the upper right corner to close the menu

**Comment by calvinckho** [comment]
@ionic/core 4.10.0 fixed it.

**Comment by liamdebeasi** [comment]
I can confirm that this issue has been resolved in Ionic 4.10.0. For any new bugs, please open a new issue. Thanks!
